vm.h

```diff
@@ -46,4 +48,22 @@ void vm_hook_opcode_handler(vm_env *, int, vm_handler);
 
 void vm_run(vm_env *env);
 
+vm_seg_info *vm_new_seg_info(char *mem, size_t sz);
```

Comment:
I'm not familiar with the internal design and history of LLVM/GCC. Actually I'm not familiar with the design of any compilter/VM... :)

The original implementation in vm.c was to return memory pointer to ELF. It suffered an issue: the strings are using host pointer. After been save to ELF and load ELF, the loaded memory has wrong pointer.

I take this internal string pointer as an internal structure that can expedite the execution. So it was not adequate to expose this directly. Instead, I implemented inflate/deflate pairs of function to convert back/forth the internal format and the external format for ELF storage.

I think this part needs a lot more design to be complete. Although I'm not familiar with mc.h, it is definitely nice to learn it (maybe later).